compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -524,111 +525,113 @@ createProjectPackageDb opts fps = do
-                              , pkgRefs)
-              let pkgIdsTopoSorted = reverse $ topSort depGraph
-              dbPathAbs <- makeAbsolute dbPath
-              projectPackageDatabaseAbs <- makeAbsolute projectPackageDatabase
-              forM_ pkgIdsTopoSorted $ \vertex -> do
-                  let ((src, templInstSrc, uid, dalf, bs), pkgId, _) =
-                          vertexToNode vertex
-                  when (uid /= primUnitId) $ do
-                      let unitIdStr = unitIdString uid
-                      let instancesUnitIdStr = "instances-" <> unitIdStr
-                      let pkgIdStr = T.unpack $ LF.unPackageId pkgId
-                      let (pkgName, mbPkgVersion) =
-                              fromMaybe (unitIdStr, Nothing) $ do
-                                  (uId, ver) <- stripInfixEnd "-" unitIdStr
-                                  guard $ all (`elem` '.' : ['0' .. '9']) ver
-                                  Just (uId, Just ver)
-                      let deps =
-                              [ unitIdString uId <.> "dalf"
-                              | ((_src, _templSrc, uId, _dalf, _bs), pId, _) <-
-                                    map vertexToNode $ reachable depGraph vertex
-                              , pkgId /= pId
-                              ]
-                      let workDir = dbPath </> unitIdStr <> "-" <> pkgIdStr
-                      createDirectoryIfMissing True workDir
-                      -- write the dalf package
-                      B.writeFile (workDir </> unitIdStr <.> "dalf") bs
-                      generateAndInstallIfaceFiles
-                          dalf
-                          src
-                          opts
-                          workDir
-                          dbPath
-                          projectPackageDatabase
-                          unitIdStr
-                          pkgIdStr
-                          pkgName
-                          mbPkgVersion
-                          deps
-
-                      unless (null templInstSrc) $
-                          generateAndInstallInstancesPkg
-                              templInstSrc
-                              opts
-                              dbPathAbs
-                              projectPackageDatabaseAbs
-                              unitIdStr
-                              instancesUnitIdStr
-                              pkgName
-                              mbPkgVersion
-                              deps
-
-       | length uniqSdkVersions <= 1 -> forM_ dars $
-            \ExtractedDar{..} -> installDar dbPath edConfFiles edDalfs edSrcs
-       | otherwise ->
+    deps <- handleSdkPackages $ filter (`notElem` basePackages) deps0
+    depsExtracted <- mapM extractDar deps
+    let uniqSdkVersions = nubSort $ filter (/= "0.0.0") $ thisSdkVer : map edSdkVersions depsExtracted
+    unless (length uniqSdkVersions <= 1 || optAllowDifferentSdks opts) $
```

Comment:
Does `optAllowDifferentSdks` still make sense? It seems like there are two cases:
1. I have a data import. In that case an SDK check doesn’t make sense.
2. I have a regular dependency. If that dependency is from a different SDK it should always be an error.

Given that I don’t see the need for manual control in either of those cases, the option doesn’t seem useful but maybe I’m missing something?

Reply:
Happy to drop it too, I check if we don't need it anymore.